src/main/java/net/fabricmc/fabric/block/Climbable.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.block;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Blocks that implement this can be climbed by LivingEntities.
+ */
+public interface Climbable {
+
+    /**
+     * Determines if the passed LivingEntity can climb this block.
+     *
+     * @param livingEntity The LivingEntity that is attempting to climb this block.
+     * @param blockState The blockstate of the ladder being climbed.
+     * @param pos The position of the block.
+     */
+    default boolean canClimb(LivingEntity livingEntity, BlockState blockState, BlockPos pos) {
```

Comment:
Notes:

1. Idea - drop the "default"; add a MixinBlock which implements the interface on all Blocks with the implementation returning false and null respectively. That way, you can cast blindly without an instanceof - not sure if the method is called enough to be a performance concern, however.
2. getDeathSuffix should be either renamed to getClimbFallDeathSuffix or moved to a more generic implementation?

Reply:
1. From my understanding, `LivingEntity#CanClimb()` is run every tick, so that method would be run every tick.
2. I think getFallDeathSuffix will do to match up with `DamageTracker#setFallDeathSuffix`.

Reply:
Resolved.